zulipterminal/model.py

```diff
@@ -914,11 +914,14 @@ def _handle_typing_event(self, event: Event) -> None:
                 else:
                     raise RuntimeError("Unknown typing event operation")
 
-    def get_invalid_recipient_emails(self, recipient_emails: List[str]
-                                     ) -> List[str]:
-
-        return [email for email in recipient_emails
-                if email not in self.user_dict]
+    def check_recipient_validity(self, recipient_name: str,
+                                 recipient_email: str,
+                                 ) -> bool:
+        if recipient_email and recipient_email in self.user_dict:
+            verified_name = self.user_dict[recipient_email]['full_name']
+            if recipient_name == verified_name:
+                return True
+        return False
```

Comment:
Currently, it's fine, though I feel it'd be nice to split this into a prior commit, where you could add relevant tests for this method as well.

Reply:
I considered doing that, but a change here in a previous commit would need me to change the function call and the calling parameters, which needs extraction of the user's name as well, which is basically what the current standalone commit is doing. :) 

For tests, I've covered the functionality in the more general tests for validation, with extensive cases that test the functioning of this method, I don't think we'd need more explicit tests for the method.

Reply:
I just noticed this comment. This method is really sort of just a helper used in one place - ideally we would have a test, but it's like an external (but encapsulated in another class) implementation function at least for now. I suspect this code will change again in the near future.

As per my review below, ideally this change could be accompanied by only a small change elsewhere. That is, in some commit we need this different function in the model to make a feature work elsewhere, but if we try and avoid doing too much in the commit at once, it'll be clearer that this change is also made just to support that change/feature.